src/ImageSharp/Formats/Tga/TgaDecoderCore.cs

```diff
@@ -29,12 +29,12 @@ internal sealed class TgaDecoderCore : IImageDecoderInternals
     /// <summary>
     /// The metadata.
     /// </summary>
-    private ImageMetadata metadata;
+    private ImageMetadata metadata = null!;
```

Comment:
Whoa! I didn't know this was possible.

Reply:
Is it a good idea? We haven't used this strategy elsewhere.

Reply:
It is an easy way. In the decoder and encoder this is not null while processing because it is initialized in the public methods. 

To make it more future proof it is probably better to not do that. 

But then we need Guards in many places or some !

Reply:
I think we should probably go the future proof way and be consistent with the other formats